git_machete/tests/functional/test_machete.py

```diff
@@ -1381,3 +1552,236 @@ def test_update_with_fork_point_specified(self) -> None:
+        self.launch_command("discover", "-y")
+        self.launch_command("github", "create-pr")
+        # ahead of origin state, push is advised and accepted
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger (diverged from origin)
+              |
+              o-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields (untracked)
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws *  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+        self.repo_sandbox.check_out('chore/fields')
+        #  untracked state (can only create pr when branch is pushed)
+        self.launch_command("github", "create-pr")
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger (diverged from origin)
+              |
+              o-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields *  PR #5
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+
+        (
+            self.repo_sandbox.check_out('hotfix/add-trigger')
+            .commit('trigger released')
+            .commit('minor changes applied')
+        )
+
+        # diverged from and newer than origin
+        self.launch_command("github", "create-pr")
+        self.assert_command(
+            ['status'],
+            """
+            master
+            |
+            o-hotfix/add-trigger *  PR #6
+              |
+              x-ignore-trailing (diverged from & older than origin)
+                |
+                o-chore/fields  PR #5
+
+            develop
+            |
+            x-allow-ownership-link (ahead of origin)
+            | |
+            | x-build-chain (untracked)
+            |
+            o-call-ws  PR #4
+              |
+              x-drop-constraint (untracked)
+            """,
+        )
+        # check against attempt to create already existing pull request
+        machete_client = cmd.MacheteClient(cli_opts, git)
+        expected_error_message = "Pull request for branch hotfix/add-trigger is already created under link www.github.com!\nPR details: PR #6 by githubuser: hotfix/add-trigger -> master"
+        with self.assertRaises(MacheteException) as e:
+            machete_client.read_definition_file()
+            machete_client.create_github_pr('hotfix/add-trigger', draft=False)
```

Comment:
Which line should raise exception? I would expect that only one line will be in the context and other, that we don't expect to raise exception will be not in the context. Otherwise we can get all sort of wrong results from the assertRaises.

Reply:
assertEqual will catch desired exception, but ok.